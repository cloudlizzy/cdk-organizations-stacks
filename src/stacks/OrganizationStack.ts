import { Account, FeatureSet, Organization, OrganizationalUnit } from '@pepperize/cdk-organizations';
import { Stack, StackProps } from 'aws-cdk-lib';
import { Construct } from 'constructs';


export interface OrganizationStackProps extends StackProps {
  organizationUnitName: string;
  accountName: string;
  accountEmail: string;
  env: {
    region: string;
    account: string;
  };
}

export class OrganizationStack extends Stack {
  constructor(scope: Construct, id: string, props: OrganizationStackProps) {
    super(scope, id, props);

    // define resources here...
    const organization = new Organization(this, 'Organization', {
      featureSet: FeatureSet.ALL,
    });

    // Create an organizational unit (OU)
    const organizationUnit = new OrganizationalUnit(this, 'OrganizationalUnit', {
      organizationalUnitName: props.organizationUnitName,
      parent: organization.root,
    });

    // Create an account
    const account = new Account(this, 'Account', {
      accountName: props.accountName,
      email: props.accountEmail,
      parent: organizationUnit,
    });

    if (account) {
    }
  }
}
