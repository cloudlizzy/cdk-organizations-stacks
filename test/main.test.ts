import { App } from 'aws-cdk-lib';
import { Template } from 'aws-cdk-lib/assertions';
import { OrganizationStack } from '../src/stacks/OrganizationStack';

test('Snapshot', () => {
  const app = new App();
  const stack = new OrganizationStack(app, 'test', {
    accountEmail: 'test@email.com',
    accountName: 'test',
    organizationUnitName: 'test',
    env: {
      account: process.env.CDK_DEFAULT_ACCOUNT as string,
      region: process.env.CDK_DEFAULT_REGION as string,
    },
  });

  const template = Template.fromStack(stack);
  expect(template.toJSON()).toMatchSnapshot();
});