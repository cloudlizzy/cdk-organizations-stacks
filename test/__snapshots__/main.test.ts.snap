// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot 1`] = `
{
  "Mappings": {
    "DefaultCrNodeVersionMap": {
      "af-south-1": {
        "value": "nodejs16.x",
      },
      "ap-east-1": {
        "value": "nodejs16.x",
      },
      "ap-northeast-1": {
        "value": "nodejs16.x",
      },
      "ap-northeast-2": {
        "value": "nodejs16.x",
      },
      "ap-northeast-3": {
        "value": "nodejs16.x",
      },
      "ap-south-1": {
        "value": "nodejs16.x",
      },
      "ap-south-2": {
        "value": "nodejs16.x",
      },
      "ap-southeast-1": {
        "value": "nodejs16.x",
      },
      "ap-southeast-2": {
        "value": "nodejs16.x",
      },
      "ap-southeast-3": {
        "value": "nodejs16.x",
      },
      "ca-central-1": {
        "value": "nodejs16.x",
      },
      "cn-north-1": {
        "value": "nodejs16.x",
      },
      "cn-northwest-1": {
        "value": "nodejs16.x",
      },
      "eu-central-1": {
        "value": "nodejs16.x",
      },
      "eu-central-2": {
        "value": "nodejs16.x",
      },
      "eu-north-1": {
        "value": "nodejs16.x",
      },
      "eu-south-1": {
        "value": "nodejs16.x",
      },
      "eu-south-2": {
        "value": "nodejs16.x",
      },
      "eu-west-1": {
        "value": "nodejs16.x",
      },
      "eu-west-2": {
        "value": "nodejs16.x",
      },
      "eu-west-3": {
        "value": "nodejs16.x",
      },
      "me-central-1": {
        "value": "nodejs16.x",
      },
      "me-south-1": {
        "value": "nodejs16.x",
      },
      "sa-east-1": {
        "value": "nodejs16.x",
      },
      "us-east-1": {
        "value": "nodejs16.x",
      },
      "us-east-2": {
        "value": "nodejs16.x",
      },
      "us-gov-east-1": {
        "value": "nodejs16.x",
      },
      "us-gov-west-1": {
        "value": "nodejs16.x",
      },
      "us-iso-east-1": {
        "value": "nodejs14.x",
      },
      "us-iso-west-1": {
        "value": "nodejs14.x",
      },
      "us-isob-east-1": {
        "value": "nodejs14.x",
      },
      "us-west-1": {
        "value": "nodejs16.x",
      },
      "us-west-2": {
        "value": "nodejs16.x",
      },
    },
  },
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "AWS679f53fac002430cb0da5b7982bd22872D164C4C": {
      "DependsOn": [
        "AWS679f53fac002430cb0da5b7982bd2287ServiceRoleC1EA0FF2",
      ],
      "Properties": {
        "Code": {
          "S3Bucket": {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "S3Key": "f9346b940b724b094a16ca051c017799995fa93df6da38a0539bf7c000fee50a.zip",
        },
        "Handler": "index.handler",
        "Role": {
          "Fn::GetAtt": [
            "AWS679f53fac002430cb0da5b7982bd2287ServiceRoleC1EA0FF2",
            "Arn",
          ],
        },
        "Runtime": {
          "Fn::FindInMap": [
            "DefaultCrNodeVersionMap",
            {
              "Ref": "AWS::Region",
            },
            "value",
          ],
        },
        "Timeout": 120,
      },
      "Type": "AWS::Lambda::Function",
    },
    "AWS679f53fac002430cb0da5b7982bd2287ServiceRoleC1EA0FF2": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "AccountCreateAccount833709C2": {
      "DeletionPolicy": "Delete",
      "DependsOn": [
        "OrganizationalUnitOrganizationProvider9D2E0DDF",
        "OrganizationalUnitTagsTagResource4B8852D8",
      ],
      "Properties": {
        "AccountName": "test",
        "Email": "test@email.com",
        "IamUserAccessToBilling": "ALLOW",
        "ImportOnDuplicate": "true",
        "ParentId": {
          "Fn::GetAtt": [
            "OrganizationalUnitOrganizationProvider9D2E0DDF",
            "Id",
          ],
        },
        "RemovalPolicy": "retain",
        "RoleName": "OrganizationAccountAccessRole",
        "ServiceToken": {
          "Fn::GetAtt": [
            "cdkorganizationsAccountProviderNestedStackcdkorganizationsAccountProviderNestedStackResourceA1C2E3D5",
            "Outputs.testcdkorganizationsAccountProviderframeworkonEventC31BCB63Arn",
          ],
        },
      },
      "Type": "Custom::Organizations_Account",
      "UpdateReplacePolicy": "Delete",
    },
    "AccountTagsTagResourceB6D57C22": {
      "DeletionPolicy": "Delete",
      "DependsOn": [
        "AccountCreateAccount833709C2",
        "OrganizationalUnitOrganizationProvider9D2E0DDF",
        "OrganizationalUnitTagsTagResource4B8852D8",
      ],
      "Properties": {
        "ResourceId": {
          "Fn::GetAtt": [
            "AccountCreateAccount833709C2",
            "AccountId",
          ],
        },
        "ServiceToken": {
          "Fn::GetAtt": [
            "cdkorganizationsTagResourceProviderNestedStackcdkorganizationsTagResourceProviderNestedStackResourceA7B8DF96",
            "Outputs.testcdkorganizationsTagResourceProviderframeworkonEventBF228F58Arn",
          ],
        },
      },
      "Type": "Custom::Organizations_TagResource",
      "UpdateReplacePolicy": "Delete",
    },
    "Organization06E16095": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "FeatureSet": "ALL",
        "ServiceToken": {
          "Fn::GetAtt": [
            "cdkorganizationsOrganizationProviderNestedStackcdkorganizationsOrganizationProviderNestedStackResourceE0751832",
            "Outputs.testcdkorganizationsOrganizationProviderframeworkonEvent9F8877C4Arn",
          ],
        },
      },
      "Type": "Custom::Organizations_Organization",
      "UpdateReplacePolicy": "Delete",
    },
    "OrganizationRootRootCustomResourceBB74F060": {
      "DeletionPolicy": "Delete",
      "DependsOn": [
        "Organization06E16095",
        "OrganizationRootRootCustomResourceCustomResourcePolicyB45F831E",
      ],
      "Properties": {
        "Create": "{"service":"Organizations","action":"listRoots","region":"us-east-1","physicalResourceId":{"responsePath":"Roots.0.Id"}}",
        "Delete": "{"service":"Organizations","action":"listRoots","region":"us-east-1"}",
        "InstallLatestAwsSdk": false,
        "ServiceToken": {
          "Fn::GetAtt": [
            "AWS679f53fac002430cb0da5b7982bd22872D164C4C",
            "Arn",
          ],
        },
        "Update": "{"service":"Organizations","action":"listRoots","region":"us-east-1","physicalResourceId":{"responsePath":"Roots.0.Id"}}",
      },
      "Type": "Custom::Organizations_Root",
      "UpdateReplacePolicy": "Delete",
    },
    "OrganizationRootRootCustomResourceCustomResourcePolicyB45F831E": {
      "DependsOn": [
        "Organization06E16095",
      ],
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": "organizations:ListRoots",
              "Effect": "Allow",
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "OrganizationRootRootCustomResourceCustomResourcePolicyB45F831E",
        "Roles": [
          {
            "Ref": "AWS679f53fac002430cb0da5b7982bd2287ServiceRoleC1EA0FF2",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "OrganizationRootTagsTagResourceCBEA7B2F": {
      "DeletionPolicy": "Delete",
      "DependsOn": [
        "Organization06E16095",
        "OrganizationRootRootCustomResourceCustomResourcePolicyB45F831E",
        "OrganizationRootRootCustomResourceBB74F060",
      ],
      "Properties": {
        "ResourceId": {
          "Fn::GetAtt": [
            "OrganizationRootRootCustomResourceBB74F060",
            "Roots.0.Id",
          ],
        },
        "ServiceToken": {
          "Fn::GetAtt": [
            "cdkorganizationsTagResourceProviderNestedStackcdkorganizationsTagResourceProviderNestedStackResourceA7B8DF96",
            "Outputs.testcdkorganizationsTagResourceProviderframeworkonEventBF228F58Arn",
          ],
        },
      },
      "Type": "Custom::Organizations_TagResource",
      "UpdateReplacePolicy": "Delete",
    },
    "OrganizationalUnitOrganizationProvider9D2E0DDF": {
      "DeletionPolicy": "Delete",
      "DependsOn": [
        "OrganizationRootRootCustomResourceCustomResourcePolicyB45F831E",
        "OrganizationRootRootCustomResourceBB74F060",
        "OrganizationRootTagsTagResourceCBEA7B2F",
      ],
      "Properties": {
        "ImportOnDuplicate": "true",
        "Name": "test",
        "ParentId": {
          "Fn::GetAtt": [
            "OrganizationRootRootCustomResourceBB74F060",
            "Roots.0.Id",
          ],
        },
        "RemovalPolicy": "retain",
        "ServiceToken": {
          "Fn::GetAtt": [
            "cdkorganizationsOrganizationalUnitProviderNestedStackcdkorganizationsOrganizationalUnitProviderNestedStackResource4FB360EE",
            "Outputs.testcdkorganizationsOrganizationalUnitProviderframeworkonEvent940556B3Arn",
          ],
        },
      },
      "Type": "Custom::Organizations_OrganizationalUnitProvider",
      "UpdateReplacePolicy": "Delete",
    },
    "OrganizationalUnitTagsTagResource4B8852D8": {
      "DeletionPolicy": "Delete",
      "DependsOn": [
        "OrganizationRootRootCustomResourceCustomResourcePolicyB45F831E",
        "OrganizationRootRootCustomResourceBB74F060",
        "OrganizationRootTagsTagResourceCBEA7B2F",
        "OrganizationalUnitOrganizationProvider9D2E0DDF",
      ],
      "Properties": {
        "ResourceId": {
          "Fn::GetAtt": [
            "OrganizationalUnitOrganizationProvider9D2E0DDF",
            "Id",
          ],
        },
        "ServiceToken": {
          "Fn::GetAtt": [
            "cdkorganizationsTagResourceProviderNestedStackcdkorganizationsTagResourceProviderNestedStackResourceA7B8DF96",
            "Outputs.testcdkorganizationsTagResourceProviderframeworkonEventBF228F58Arn",
          ],
        },
      },
      "Type": "Custom::Organizations_TagResource",
      "UpdateReplacePolicy": "Delete",
    },
    "cdkorganizationsAccountProviderNestedStackcdkorganizationsAccountProviderNestedStackResourceA1C2E3D5": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "TemplateURL": {
          "Fn::Join": [
            "",
            [
              "https://s3.",
              {
                "Ref": "AWS::Region",
              },
              ".",
              {
                "Ref": "AWS::URLSuffix",
              },
              "/",
              {
                "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
              },
              "/8f85e8dcb95e073c0f165dad4cdfe6e9dce286b53141e1ab3656a1258f260932.json",
            ],
          ],
        },
      },
      "Type": "AWS::CloudFormation::Stack",
      "UpdateReplacePolicy": "Delete",
    },
    "cdkorganizationsOrganizationProviderNestedStackcdkorganizationsOrganizationProviderNestedStackResourceE0751832": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "TemplateURL": {
          "Fn::Join": [
            "",
            [
              "https://s3.",
              {
                "Ref": "AWS::Region",
              },
              ".",
              {
                "Ref": "AWS::URLSuffix",
              },
              "/",
              {
                "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
              },
              "/355546a0bcdf7a85979d1d0d844f1a4b034b36c31f2111eedc6fb2f8e1027149.json",
            ],
          ],
        },
      },
      "Type": "AWS::CloudFormation::Stack",
      "UpdateReplacePolicy": "Delete",
    },
    "cdkorganizationsOrganizationalUnitProviderNestedStackcdkorganizationsOrganizationalUnitProviderNestedStackResource4FB360EE": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "TemplateURL": {
          "Fn::Join": [
            "",
            [
              "https://s3.",
              {
                "Ref": "AWS::Region",
              },
              ".",
              {
                "Ref": "AWS::URLSuffix",
              },
              "/",
              {
                "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
              },
              "/3117045fd5c4700f562ea4a9beff5d6f32fefc42a6908ffff06830fde2ac7e44.json",
            ],
          ],
        },
      },
      "Type": "AWS::CloudFormation::Stack",
      "UpdateReplacePolicy": "Delete",
    },
    "cdkorganizationsTagResourceProviderNestedStackcdkorganizationsTagResourceProviderNestedStackResourceA7B8DF96": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "TemplateURL": {
          "Fn::Join": [
            "",
            [
              "https://s3.",
              {
                "Ref": "AWS::Region",
              },
              ".",
              {
                "Ref": "AWS::URLSuffix",
              },
              "/",
              {
                "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
              },
              "/d6ca4084620f79c6fd7ab880f4f81332c6c92ab799e40df07910c2cc405d24e1.json",
            ],
          ],
        },
      },
      "Type": "AWS::CloudFormation::Stack",
      "UpdateReplacePolicy": "Delete",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
